sdk/core/Azure.Core.TestFramework/src/UseSyncMethodsInterceptor.cs

```diff
@@ -186,31 +193,42 @@ private static MethodInfo GetMethod(IInvocation invocation, string nonAsyncMetho
             // only comparing the generic type or count and it's only enough
             // for the cases we have today.
             static Type GenericDef(Type t) => t.IsGenericType ? t.GetGenericTypeDefinition() : t;
-            MethodInfo GetMethodSlow() =>
-                invocation.TargetType
-                    // Start with all methods that have the right name
-                    .GetMethods(flags).Where(m => m.Name == nonAsyncMethodName)
-                    // Check if their type parameters have the same generic
-                    // type definitions (i.e., if I invoked
-                    // GetAsync<Model>(Wrapper<Model>) we want that to match
-                    // with Get<T>(Wrapper<T>)
-                    .Where(m =>
-                        m.GetParameters().Select(p => GenericDef(p.ParameterType))
-                        .SequenceEqual(types.Select(GenericDef)))
-                    // Check if they have the same number of type arguments
-                    // (all of our cases today either specialize on 0 or 1 type
-                    // argument for the static or dynamic user schema approach)
-                    .Where(m =>
+            static bool AssignableFromGenericOrEqual(Type t1, Type t2) => t1.IsGenericParameter ? t1.GetGenericParameterConstraints().Any(c => c.IsAssignableFrom(t2)) : t1 == t2;// t.IsAssignableFrom(t2);
+            MethodInfo GetMethodSlow()
+            {
+                var methods = invocation.TargetType
```

Comment:
I wonder if we can move from trying to match generic arguments to trying to closing MethodInfo around the same set of arguments that initial method has and doing a simple parameter comparison.